201-timeseriesinsights-environment-with-eventhub/azuredeploy.json

```diff
@@ -194,16 +208,50 @@
           },
           "tags": "[variables('eventSourceTagsValue')]",
           "dependsOn": [
-            "[concat('Microsoft.TimeSeriesInsights/environments/',parameters('environmentName'))]",
+            "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]",
             "[concat('Microsoft.EventHub/namespaces/', parameters('eventHubNamespaceName'), '/eventHubs/', parameters('eventHubName'), '/consumerGroups/', parameters('consumerGroupname'))]"
           ]
         }
       ]
+    },
+    {
+      "condition": "[if(equals(first(parameters('accessPolicyReaderObjectIds')), 'thisArrayIsReallyEmpty'), bool('false'), bool('true'))]",
+      "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+      "name": "[concat(parameters('environmentName'), '/', 'readerAccessPolicy', copyIndex())]",
+      "copy": { 
+        "name": "accessPolicyReaderCopy", 
+        "count": "[if(equals(parameters('accessPolicyReaderObjectIds'), 0), 1, length(parameters('accessPolicyReaderObjectIds')))]"
+      },
+      "apiVersion": "2017-11-15",
+      "properties": {
+        "principalObjectId": "[parameters('accessPolicyReaderObjectIds')[copyIndex()]]",
+        "roles": ["Reader"]
+      },
+      "dependsOn": [
+        "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]"
+      ]
+    },
+    {
+      "condition": "[if(equals(first(parameters('accessPolicyContributorObjectIds')), 'thisArrayIsReallyEmpty'), bool('false'), bool('true'))]",
```

Comment:
same as above

Reply:
I think I can do this (I haven't tested it yet) if I go with the alternative of conditionally setting copy.lenth to 1 if 'accessPolicyContributorObjectIds' is empty. But I honestly think someone who sees that is just going to think it's a template authoring bug, where my current workaround is a little more obvious. I could put something in the description indicating why the default value is ["thisArrayIsReallyEmpty"]. But this would bleed through to the deployment in the portal. Do you have any real preference for either of the two workarounds?
(I wish I could just add a comment in the template.)